# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1205169010 14400
#      Mon Mar 10 13:10:10 2008 -0400
# Node ID 9669b69e34ef0105bd193575e6bb0556c9a5d4bb
# Parent  755bf7ecc29b4b4367c103ece4c80908abe5cb31
define mxLogical as unsigned char instead of int

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-03-10  John W. Eaton  <jwe@octave.org>
+
+	* mxarray.h.in (mxLogical): Use unsigned char instead of int.
+	From Antwerpen, G. (Gert) van <gert.vanantwerpen@tno.nl>.
+
 2008-03-07  John W. Eaton  <jwe@octave.org>
 
 	* ov-struct.cc (octave_struct::print_raw): Don't print contents fo
 	struct arrays that have more than one element.
 
 2008-03-06  David Bateman  <dbateman@free.fr>
 
 	* DLD-FUNCTIONS/matrix_type.cc (Fmatrix_type): Document that
diff --git a/src/mxarray.h.in b/src/mxarray.h.in
--- a/src/mxarray.h.in
+++ b/src/mxarray.h.in
@@ -70,17 +70,17 @@ typedef enum
     mxINT32_CLASS,
     mxUINT32_CLASS,
     mxINT64_CLASS,
     mxUINT64_CLASS,
     mxFUNCTION_CLASS
   }
   mxClassID;
 
-typedef int mxLogical;
+typedef unsigned char mxLogical;
 
 /* typedef Uint16 mxChar; */
 typedef char mxChar;
 
 /* 
  * FIXME? Mathworks says these should be size_t on 64-bit system and when
  * mex is used with the -largearraydims flag, but why do that? Its better
  * to conform to the same indexing as the rest of Octave
